project/src/mobs/MobSpawner.gd

```diff
@@ -0,0 +1,42 @@
+extends Node2D
+class_name MobSpawner
+
+var RandomUtils = preload("res://src/RandomUtils.gd").new()
+
+export var spawn_group_size: int = 8
+export var mob_scene: PackedScene
+export var MAX_MOB_COUNT := 50
+
+onready var path: Path2D = $SpawnerPath2D
+onready var curve: Curve2D = path.curve
+onready var num_points = len(curve.get_baked_points())
+onready var mob_parent = get_node("/root/")
+onready var timer = $SpawnInterval
+
+func _ready():
+	spawn_mob_group()
+
+func get_mob_count():
+	return len(get_tree().get_nodes_in_group("mobs"))
+
+func spawn_mob_group():
+	for i in spawn_group_size:
+		spawn_mob()
+
+func is_below_max_mob_count():
+	return get_mob_count() < MAX_MOB_COUNT
+
+func instance_mob(position: Vector2) -> Mob:
+	var mob = mob_scene.instance()
+	mob.global_position = position
+	return mob
+
```

Comment:
spawn_mob needs a time component so that every mob isn't spawned in at the same time. The time component would also be a naturally limiting factor on the number of onscreen mobs. We could add that later, but I feel as it looks now we aren't quite at expected behavior for the spawner. If your intent is to get this on main asap that is also fine rn.

Reply:
This looks good for now. Any changes needed are nb.